src/script/cpp_api/s_player.cpp

```diff
@@ -156,7 +156,9 @@ void ScriptApiPlayer::on_joinplayer(ServerActiveObject *player)
 	lua_getfield(L, -1, "registered_on_joinplayers");
 	// Call callbacks
 	objectrefGetOrCreate(L, player);
-	runCallbacks(1, RUN_CALLBACKS_MODE_FIRST);
+	if(last_login != -1)
+		lua_pushinteger(L, last_login);
+	runCallbacks(last_login != -1 ? 2 : 1, RUN_CALLBACKS_MODE_FIRST);
```

Comment:
Please push `nil` and pass two values, rather than leaving it dynamic. This way it would be possible in mods to check whether the server supports this feature (`{...}` generic argument capture).